projects/ngx-avatar/src/lib/avatar.component.ts

```diff
@@ -153,18 +151,34 @@ export class AvatarComponent implements OnChanges, OnDestroy {
    *
    * memberOf AvatarComponent
    */
-  public fetchAvatarSource(event?: any): void {
-    const avatarSource = this.sources[this.currentSource];
-    if (!avatarSource) {
+  public fetchAvatarSource(): void {
+    const previousSource = this.sources[this.currentIndex];
+    if (previousSource) {
+      this.avatarService.markSourceAsFailed(previousSource);
+    }
+
+    const source = this.findNextSource();
+    if (!source) {
       return;
     }
-    if (this.avatarService.isTextAvatar(avatarSource.sourceType)) {
-      this.buildTextAvatar(avatarSource);
+
+    if (this.avatarService.isTextAvatar(source.sourceType)) {
+      this.buildTextAvatar(source);
       this.avatarSrc = null;
     } else {
-      this.buildImageAvatar(avatarSource);
+      this.buildImageAvatar(source);
+    }
+  }
```

Comment:
`findNextSource` centralize the search for an available source and is responsible to maintain `this.currentIndex` accordingly. So only call that method is enough, and `this.currentIndex` should not be modified anywhere else.

Reply:
Personally I don't like this, but I'm ok with this.

Reply:
The whole thing could be extracted into a new class SourceCollection that would handle adding, removing, and fetching next available one. But that's probably a bit out of scope, though it would be nice...